hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/Route.java

```diff
@@ -51,10 +92,23 @@ public void configure() {
             //dispatch the message based on the receiving application code and message type
             .choice()
 	            //sending message to pharmaNet
-	            .when(simple("${in.header.receivingApp} == {{pharmanet-endpoint}}"))
-                    .log("The pharmaNet endpoint(${in.header.receivingApp}) is reached and message will be sent to PharmaNet webservices")
-                    .setBody(simple(SampleMessages.pnpResponseMessage))
-	            
+            	.when(header("receivingApp").isEqualTo(Util.RECEIVING_APP_PNP))
+                	.log("Message identified as PharmaNet message. Preparing message for PharmaNet.")
+            		.to("log:HttpLogger?level=DEBUG&showBody=true&multiline=true")
+            		.setBody(body().regexReplaceAll("\r\n","\r"))
+                    .setBody().method(new Base64Encoder())
+		            .setBody().method(new ProcessV2ToPharmaNetJson()).id("ProcessV2ToPharmaNetJson")
+		            .log("Sending to Pharmanet")
+		            .removeHeader(Exchange.HTTP_URI) //clean this header as it has been set in the "from" section
+		            .setHeader(Exchange.CONTENT_TYPE, constant("application/json"))		            
+		            .setHeader("CamelHttpMethod", constant("POST"))
+		            .setHeader("Authorization", simple(basicToken))
```

Comment:
<low priority> We have a keyword for "Authorization" in ValidateAccessToken. May be we should create a Keywords.java and have it all there.. like "Authorization", "ssl", "ssl2", "CamelHttpMethod", "Post", "Basic"

Reply:
Extracted all Strings to constants. Added common constants to Util.java and replaced the strings in hn-secure.